Document FOCUS conformance (#909)
microsoft/finops-toolkit: docs-mslearn/focus/conformance-summary.md

| AvailabilityZone is not available in cost datasets and the column will not be present.                                                                                                  | Join with resource details.                   | AvailabilityZone<br>AZ1                                                |

<br>

## Incorrect data

| Issue                                                                                                       | Workaround                                           | Columns / Requirements                                                                                                                                      |
| ----------------------------------------------------------------------------------------------------------- | ---------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | Date columns all follow the ISO 8601 standard, but do not include seconds (e.g., “2024-01-01T00:00Z”). | Replace “T00:00Z” with “T00:00:00Z”. Do not add “:00” without confirming seconds are not present. This will be resolved in “1.0r2”. | BillingPeriodStart, BillingPeriodEnd, ChargePeriodStart, ChargePeriodEnd, x_BillingExchangeRateDate, x_ServicePeriodEnd, x_ServicePeriodStart<br>DTF5 |

Author: ```suggestion
| Date columns all follow the ISO 8601 standard, but do not include seconds (e.g., “2024-01-01T00:00Z”). | Replace “T00:00Z” with “T00:00:00Z”. Do not add “:00” without confirming seconds are not present. This will be resolved in a new FOCUS "1.0r2” dataset version. | BillingPeriodStart, BillingPeriodEnd, ChargePeriodStart, ChargePeriodEnd, x_BillingExchangeRateDate, x_ServicePeriodEnd, x_ServicePeriodStart<br>DTF5 |
```

| Some columns can be "-2" or “Unassigned” when there is no value.                                            | Replace "-2" and “Unassigned” with null.             | SkuPriceId, SubAccountName, x_AccountId, x_AccountName, x_AccountOwnerId, x_InvoiceSectionId, x_InvoiceSectionName, x_PricingUnitDescription<br>NH1-3, SAN3 |

<br>

## Inconsistent data

| Issue                                                                                                               | Workaround                                                                                                                  | Columns / Requirements                          |
| ------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------- |

> | SkuPriceId does not identify a single price and does not have an equivalent in the price sheet.                     | Join with the price sheet using x_SkuMeterId, x_SkuOrderId, x_SkuOffer, and CommitmentDiscountType.                         | SkuPriceId<br>SkPI4                             |

Author: ```suggestion
| SkuPriceId does not identify a single price and does not have an equivalent in the price sheet.                     | For EA accounts, join with the price sheet using `x_SkuMeterId`/`MeterId`, `x_SkuOfferId`/`OfferID`, `x_SkuTerm`/`Term`, and `CommitmentDiscountType`/`PriceType`. For MCA accounts, join with the price sheet using `SkuPriceId` and a concatenated “{ProductId}_{SkuId}_{MeterId” string, `x_SkuTerm`/`Term`, and `CommitmentDiscountType`/`PriceType`. Note MCA accounts also require the tier, which is not currently available in any cost details dataset. | SkuPriceId<br>SkPI4                             |
```

| AvailabilityZone is not available in cost datasets and the column will not be present.                                                                                                  | Join with resource details.                   | AvailabilityZone<br>AZ1                                                |

<br>

## Incorrect data

| Issue                                                                                                       | Workaround                                           | Columns / Requirements                                                                                                                                      |
| ----------------------------------------------------------------------------------------------------------- | ---------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | BillingPeriodEnd has a known bug where the time is "23:59:59.999" when it should be "00:00:00".             | Remove the time element or update it to be midnight. | BillingPeriodEnd<br>BPE2.2                                                                                                                                  |
> | PricingQuantity and ContractedCost use the wrong scale and are incorrect for Enterprise Agreement accounts. | Multiply by x_PricingBlockSize.                      | ContractedCost, PricingQuantity<br>PQ5, CnC3, CnC5, LC3                                                                                                     |

Author: ```suggestion
| PricingQuantity and ContractedCost use the wrong scale and are incorrect for Enterprise Agreement accounts. | Multiply by x_PricingBlockSize.                      | ContractedCost, PricingQuantity<br>PQ5, CnC3, CnC5, LC3                                                                                                     |
| BillingPeriodEnd has a known bug where the time is "23:59:59.999" when it should be "00:00:00".             | Remove the time element or update it to be midnight. | BillingPeriodEnd<br>BPE2.2                                                                                                                                  |
| Date columns all follow the ISO 8601 standard, but do not include seconds (e.g., “2024-01-01T00:00Z”). | Replace “T00:00Z” with “T00:00:00Z”. Do not add “:00” without confirming seconds are not present. This will be resolved in “1.0r2”. | BillingPeriodStart, BillingPeriodEnd, ChargePeriodStart, ChargePeriodEnd, x_BillingExchangeRateDate, x_ServicePeriodEnd, x_ServicePeriodStart<br>DTF5 |
```

---
title: Conformance summary
description: General information about the FOCUS dataset including the data generator, schema version, and columns included in the dataset.
author: bandersmsft
ms.author: banders
ms.date: 07/15/2024
ms.topic: reference
ms.service: finops
ms.reviewer: micflan
---

<!-- markdownlint-disable-next-line MD025 -->
# FOCUS conformance summary

This document summarizes the known conformance gaps for the latest FOCUS 1.0 dataset in Microsoft Cost Management compared to the FOCUS 1.0 specification. To learn more about FOCUS, refer to the [FOCUS overview](./what-is-focus.md).

<br>

## Overall conformance

> The Microsoft Cost Management FOCUS 1.0 dataset has a conformance score of **95%**.

Author: ```suggestion
The Microsoft Cost Management FOCUS 1.0 dataset has a conformance score of **96%**.
```

| AvailabilityZone is not available in cost datasets and the column will not be present.                                                                                                  | Join with resource details.                   | AvailabilityZone<br>AZ1                                                |

<br>

## Incorrect data

| Issue                                                                                                       | Workaround                                           | Columns / Requirements                                                                                                                                      |
| ----------------------------------------------------------------------------------------------------------- | ---------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | PricingQuantity and ContractedCost use the wrong scale and are incorrect for Enterprise Agreement accounts. | Multiply by x_PricingBlockSize.                      | ContractedCost, PricingQuantity<br>PQ5, CnC3, CnC5, LC3                                                                                                     |
> | BillingPeriodEnd has a known bug where the time is "23:59:59.999" when it should be "00:00:00".             | Remove the time element or update it to be midnight. | BillingPeriodEnd<br>BPE2.2                                                                                                                                  |

Author: ```suggestion
```
